search/routes/api/classic.py

```diff
@@ -28,18 +28,16 @@ def query() -> Response:
     # requested = request.accept_mimetypes.best_match([JSON, ATOM_XML])
     # if requested == ATOM_XML:
     #     return serialize.as_atom(data), status, headers
-    response = serialize.as_json(data['results'], query=data['query'])
-    response.status_code = status_code
-    response.headers.extend(headers)
-    return response
+    response_data = serialize.as_atom(data['results'], query=data['query'])
+    return response_data, status_code, headers # type: ignore
 
 
 @blueprint.route('<arxiv:paper_id>v<string:version>', methods=['GET'])
 @scoped(required=scopes.READ_PUBLIC)
 def paper(paper_id: str, version: str) -> Response:
     """Document metadata endpoint."""
+
+    # TODO: Investigate if this method should be removed
     data, status_code, headers = api.paper(f'{paper_id}v{version}')
-    response = serialize.as_json(data['results'])
-    response.status_code = status_code
-    response.headers.extend(headers)
-    return response
+    response_data = serialize.as_json(data['results'])
+    return response_data, status_code, headers # type: ignore
```

Comment:
Same here (wrong type -> regression). Also, shouldn't we be sticking to Atom/XML for the classic API?